cosmwasm/packages/wasmi-runtime/src/wasm/runtime/contract.rs

```diff
@@ -110,6 +114,36 @@ impl ContractInstance {
         let ptr_to_region_in_wasm_vm = self.allocate(buffer.len() as u32)?;
         self.write_to_allocated_memory(buffer, ptr_to_region_in_wasm_vm)
     }
+
+    /// Track gas used inside wasmi
+    fn use_gas(&mut self, gas_amount: u64) -> Result<(), WasmEngineError> {
```

Comment:
At this point wouldn't we want to create a `GasTracker` or something to handle all these?

Reply:
we definitely could. should be really simple, so if you want me to, just say "do it" :)

Reply:
Probably not worth the effort right now, just keep things like this in mind next time

Reply:
alright :+1: